### DISCUSSION TITLE
Getting Started

### DISCUSSION THREAD

**Discussion by sa-webb** [urql-graphql/urql#1056] [Q&A] [answered]
Hello, I am trying to get started with URQL and the fullstack tutorial https://www.howtographql.com/react-urql/ seems to be out of date because I cannot seem to execute a basic query or mutation without receiving `TypeError: Cannot read property 'executeQuery' of undefined or TypeError: Cannot read property 'executeMutation' of undefined.` 

Can anyone help me execute these very simple operations? There does not seem to be any helpful resources. Thanks

**Comment by JoviDeCroock** [comment, marked as answer]
What versions are you using within your application, it feels like there's something wrong with `urql` being resolved since that implies there's no `client` in scope.

Do you have a `Provider` in your React tree?

**Reply by sa-webb** [reply]
I'm using "urql": "^1.10.1" and "@urql/exchange-graphcache": "^3.1.4"

**Reply by sa-webb** [reply]
Also, yes I have a Provider in my React Tree but i haven't used CRA in a while could StrictMode cause this issue?

**Reply by sa-webb** [reply]
`
import { Provider, Client, dedupExchange, fetchExchange } from "urql";
import { cacheExchange } from "@urql/exchange-graphcache"

const cache = cacheExchange({})

const client = new Client({
  url: "http://localhost:4000",
  exchanges: [dedupExchange, cache, fetchExchange],
});

ReactDOM.render(
  <React.StrictMode>
    <Provider client={client}>
      <App />
    </Provider>
  </React.StrictMode>,
  document.getElementById("root")
);
`

**Reply by JoviDeCroock** [reply]
I just updated the solved version to latest for urql, ... and it all seems to work correctly. StrictMode in itself shouldn't be an issue either, this feels kind off odd maybe some dependency got confused and duplicated urql somewhere... Might want to try `yarn-deduplicate` or something similar to see if there's a duplicate somewhere.

**Reply by sa-webb** [reply]
Okay thank you so much for your help!

**Reply by JoviDeCroock** [reply]
No problem, do ping here again if the problem persists, I'll look deeper into it
